src/many-client/src/client/ledger.rs

```diff
@@ -0,0 +1,13 @@
+use many_client_macros::many_client;
+use many_modules::ledger::{BalanceArgs, BalanceReturns, InfoReturns, SendArgs};
+pub use many_types::ledger::{Symbol, TokenAmount};
+
+#[many_client(
```

Comment:
It's a great start, but I found the way to declare methods confusing and easy to mistake. I would have preferred something that's a bit more rust-like for declaring the API. Something like:

```rust
#[many_client]
pub trait LedgerClientTrait {
  fn info(&self) -> Result<InfoReturns, ManyError>;
  fn balance(&self, args: BalanceArgs) -> Result<BalanceReturns, ManyError>;
  fn send(&self, args: SendArgs) -> Result<(), ManyError>;
}

pub struct LedgerClient(ManyClient);

impl LedgerClient {
  pub fn new(client: ManyClient) -> Self {
    Self(client)
  }
}

#[many_client]
impl LedgerClientTrait for LedgerClient {}
```

Or something similar. What I really want is the `fn balance(&self, args: BalanceArgs) -> Result<BalanceReturns, ManyError>` part which is a bit more code, but less confusing to readers and IDEs.

What do you think?

Reply:
@hansl I like the idea. I think the first `many_client` in that block is unnecessary, but otherwise it's seems like a nice interface.

Reply:
@hansl I like the idea of that proposal, but I think the interface would look something more like this:

```rust
#[many_client(LedgerClient)]
trait AnyTraitName {
  fn info(&self) -> Result<InfoReturns, ManyError>;
...
}

struct LedgerClient(ManyClient);
```

And then it expands to a normal impl (instead of a trait impl)